src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPDocumentMappingProvider.cs

```diff
@@ -39,45 +39,123 @@ public DefaultLSPDocumentMappingProvider(LSPRequestInvoker requestInvoker, LSPDo
             _documentManager = documentManager;
         }
 
-        public async override Task<RazorMapToDocumentRangeResponse> MapToDocumentRangeAsync(RazorLanguageKind languageKind, Uri razorDocumentUri, Range projectedRange, CancellationToken cancellationToken)
+        public async override Task<RazorMapToDocumentRangesResponse> MapToDocumentRangesAsync(RazorLanguageKind languageKind, Uri razorDocumentUri, Range[] projectedRanges, CancellationToken cancellationToken)
         {
             if (razorDocumentUri is null)
             {
                 throw new ArgumentNullException(nameof(razorDocumentUri));
             }
 
-            if (projectedRange is null)
+            if (projectedRanges is null)
             {
-                throw new ArgumentNullException(nameof(projectedRange));
+                throw new ArgumentNullException(nameof(projectedRanges));
             }
 
-            var mapToDocumentRangeParams = new RazorMapToDocumentRangeParams()
+            var mapToDocumentRangeParams = new RazorMapToDocumentRangesParams()
             {
                 Kind = languageKind,
                 RazorDocumentUri = razorDocumentUri,
-                ProjectedRange = new Range()
```

Comment:
So was the old re-building of the range not needed?

Reply:
Yeah I don't remember why I did that. I was probably being dumb.